app/models/lead_provider.rb

```diff
@@ -3,6 +3,7 @@
 class LeadProvider < ApplicationRecord
   has_many :partnerships
   has_many :schools, through: :partnerships
+  has_many :lead_provider_profiles
```

Comment:
Hmm, I'm not sure that sounds right to me.

`has_many :users, through: lead_provider_profiles` sounds a bit more like what I would expect. 

Mainly because profiles mean very little to the lead provider - almost all the operations I can think of relate to users rather than their profiles.